File: core/python/lagrange.py
def lagrange_at_zero(points, p):
    secret = 0
    for i, (xi, yi) in enumerate(points):
        num = 1
        den = 1
        for j, (xj, _) in enumerate(points):
            if i != j:
                num = (num * (-xj)) % p
                den = (den * (xi - xj)) % p
        inv_den = pow(den, -1, p)
        secret = (secret + yi * num * inv_den) % p
    return secret

File: core/javascript/lagrange.js
function lagrangeInterpolateZero(points, field) {
    let secret = 0n;

    for (let i = 0; i < points.length; i++) {
        const [xi, yi] = points[i];
        let num = 1n;
        let den = 1n;

        for (let j = 0; j < points.length; j++) {
            if (i !== j) {
                const [xj] = points[j];
                num = (num * (-xj)) % field;
                den = (den * (xi - xj)) % field;
            }
        }

        const invDen = modInverseBigInt(den, field);
        const li = (num * invDen) % field;
        secret = (secret + yi * li) % field;
    }

    return (secret + field) % field;
}

module.exports = {
    lagrangeInterpolateZero
};
